jujugui/static/gui/src/app/components/basic-table/basic-table.js

```diff
@@ -0,0 +1,112 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/** Basic table React component used to display data in a table structure. */
+class BasicTable extends React.Component {
+  constructor() {
+    super();
+    this.columnClasses = [
```

Comment:
I think that would be fine (without knowing much about CSS grids), but I think this is OK too, as we'd still have to have classes like that to set the different widths and this grid system is used across the company and maintained (and already has support for mobile etc.)